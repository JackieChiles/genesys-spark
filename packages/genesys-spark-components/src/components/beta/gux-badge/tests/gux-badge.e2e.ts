import { newSparkE2EPage, a11yCheck } from '../../../../test/e2eTestUtils';

describe('gux-badge-beta', () => {
  describe('#render', () => {
    [
      '<gux-badge-beta>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="info">Badge</gux-badge-beta>',
      '<gux-badge-beta accent="success">Badge</gux-badge-beta>',
      '<gux-badge-beta accent="warn">Badge</gux-badge-beta>',
      '<gux-badge-beta accent="error">Badge</gux-badge-beta>',
      '<gux-badge-beta accent="inherit">Badge</gux-badge-beta>',
      '<gux-badge-beta bold>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="info">Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="success">Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="warn">Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="error">Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="inherit">Badge</gux-badge-beta>',
      '<gux-badge-beta><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="info"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="success"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="warn"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="error"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta accent="inherit"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="info"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="success"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="warn"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="error"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold accent="inherit"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-badge-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });

    [
      '<gux-badge-beta>Badge</gux-badge-beta>',
      '<gux-badge-beta color="neutral">Badge</gux-badge-beta>',
      '<gux-badge-beta color="green">Badge</gux-badge-beta>',
      '<gux-badge-beta color="yellow">Badge</gux-badge-beta>',
      '<gux-badge-beta color="red">Badge</gux-badge-beta>',
      '<gux-badge-beta color="inherit">Badge</gux-badge-beta>',
      '<gux-badge-beta bold>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="neutral">Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="green">Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow">Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="red">Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="inherit">Badge</gux-badge-beta>',
      '<gux-badge-beta><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta color="neutral"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta color="green"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta color="red"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta color="inherit"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="neutral"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="green"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="red"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>',
      '<gux-badge-beta bold color="inherit"><gux-icon icon-name="subtract" decorative></gux-icon>Badge</gux-badge-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (deprecated) (${
        index + 1
      })`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-badge-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
