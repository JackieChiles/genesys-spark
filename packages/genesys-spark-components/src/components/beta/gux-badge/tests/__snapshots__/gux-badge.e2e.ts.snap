// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-badge-beta #render should render component as expected (1) 1`] = `"<gux-badge-beta hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (2) 1`] = `"<gux-badge-beta accent=\\"info\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (3) 1`] = `"<gux-badge-beta accent=\\"success\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (4) 1`] = `"<gux-badge-beta accent=\\"warn\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (5) 1`] = `"<gux-badge-beta accent=\\"error\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (6) 1`] = `"<gux-badge-beta accent=\\"inherit\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (7) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (8) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"info\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (9) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"success\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (10) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"warn\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (11) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"error\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (12) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"inherit\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (13) 1`] = `"<gux-badge-beta hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (14) 1`] = `"<gux-badge-beta accent=\\"info\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (15) 1`] = `"<gux-badge-beta accent=\\"success\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (16) 1`] = `"<gux-badge-beta accent=\\"warn\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (17) 1`] = `"<gux-badge-beta accent=\\"error\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (18) 1`] = `"<gux-badge-beta accent=\\"inherit\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (19) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (20) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"info\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (21) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"success\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (22) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"warn\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (23) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"error\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (24) 1`] = `"<gux-badge-beta bold=\\"\\" accent=\\"inherit\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (1) 1`] = `"<gux-badge-beta hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (2) 1`] = `"<gux-badge-beta color=\\"neutral\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (3) 1`] = `"<gux-badge-beta color=\\"green\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (4) 1`] = `"<gux-badge-beta color=\\"yellow\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (5) 1`] = `"<gux-badge-beta color=\\"red\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (6) 1`] = `"<gux-badge-beta color=\\"inherit\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (7) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (8) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"neutral\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (9) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"green\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (10) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"yellow\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (11) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"red\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (12) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"inherit\\" hydrated=\\"\\">Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (13) 1`] = `"<gux-badge-beta hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (14) 1`] = `"<gux-badge-beta color=\\"neutral\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (15) 1`] = `"<gux-badge-beta color=\\"green\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (16) 1`] = `"<gux-badge-beta color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (17) 1`] = `"<gux-badge-beta color=\\"red\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (18) 1`] = `"<gux-badge-beta color=\\"inherit\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (19) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (20) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"neutral\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (21) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"green\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (22) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (23) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"red\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (deprecated) (24) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"inherit\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>Badge</gux-badge-beta>"`;
